rest/route/middleware.go

```diff
@@ -170,6 +163,39 @@ func (m *projectAdminMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Reque
 	next(rw, r)
 }
 
+// This middleware is more restrictive than checkProjectAdmin, as branch admins do not have access
+func NewRepoAdminMiddleware(sc data.Connector) gimlet.Middleware {
```

Comment:
I created this for use with repo branches in the future; I thought about just using the existing project middleware, and the current ctx.ProjectRef but I worry that will result in weird edge cases in the future; since we're creating new routes for repo configurations a new middleware seems like the right move.

Reply:
it's a bit fuzzy in my mind if this is the best route. I'm assuming that there will be some routes where we want to check branch-level permission and some where we want to check project-level. So in those routes, how will we know if the identifier is a project or branch?

Reply:
Maybe I misunderstand the question. We'll never use the same route to see branch-level and to see repo-level, we'll create new routes for repos. so GET /repo_ref will use the new middleware, while GET /project_ref will use the old middleware.

Reply:
Sorry where this question comes from is thinking forward to where this will be used. You're relying on the RepoRef being in the project context so that you know what repo to check security for, but how does that get populated in the first place?

Reply:
Oh 🤦‍♀️  I wrote this but I guess I never committed it.

Reply:
Feel free to ping me on slack if this still doesn't make sense or if I'm missing something; I don't usually go this deep into middleware.